target/linux/ipq806x/patches-5.10/096-34-drivers-net-dsa-qca8k-add-support-for-multi-cpu-supp.patch

```diff
@@ -0,0 +1,119 @@
+From 662f172a947799725ecf07f7e64907ca66ece708 Mon Sep 17 00:00:00 2001
+From: Ansuel Smith <ansuelsmth@gmail.com>
+Date: Sat, 16 Oct 2021 00:20:35 +0200
+Subject: [PATCH] net: dsa: qca8k: add support for multiple-cpu port
+
+Add support for multiple-cpu port.
+
+Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
+---
+ drivers/net/dsa/qca8k.c | 65 +++++++++++++++++++++++++++++++++++++++--
+ drivers/net/dsa/qca8k.h |  1 +
+ 2 files changed, 63 insertions(+), 3 deletions(-)
+
+diff --git a/drivers/net/dsa/qca8k.c b/drivers/net/dsa/qca8k.c
+index ea7f12778922..e1d3b1084ba5 100644
+--- a/drivers/net/dsa/qca8k.c
++++ b/drivers/net/dsa/qca8k.c
+@@ -1152,9 +1152,9 @@ qca8k_setup(struct dsa_switch *ds)
+ 	 */
+ 	ret = qca8k_write(priv, QCA8K_REG_GLOBAL_FW_CTRL1,
+ 			  BIT(cpu_port) << QCA8K_GLOBAL_FW_CTRL1_IGMP_DP_S |
+-			  BIT(cpu_port) << QCA8K_GLOBAL_FW_CTRL1_BC_DP_S |
+-			  BIT(cpu_port) << QCA8K_GLOBAL_FW_CTRL1_MC_DP_S |
+-			  BIT(cpu_port) << QCA8K_GLOBAL_FW_CTRL1_UC_DP_S);
++			  dsa_cpu_ports(ds) << QCA8K_GLOBAL_FW_CTRL1_BC_DP_S |
++			  dsa_cpu_ports(ds) << QCA8K_GLOBAL_FW_CTRL1_MC_DP_S |
++			  dsa_cpu_ports(ds) << QCA8K_GLOBAL_FW_CTRL1_UC_DP_S);
+ 	if (ret)
+ 		return ret;
+ 
+@@ -1162,6 +1162,8 @@ qca8k_setup(struct dsa_switch *ds)
+ 	 * Configure specific switch configuration for ports
+ 	 */
+ 	for (i = 0; i < QCA8K_NUM_PORTS; i++) {
++		struct dsa_port *dp = dsa_to_port(ds, i);
++
+ 		/* CPU port gets connected to all user ports of the switch */
+ 		if (dsa_is_cpu_port(ds, i)) {
+ 			ret = qca8k_rmw(priv, QCA8K_PORT_LOOKUP_CTRL(i),
+@@ -1172,6 +1172,8 @@ qca8k_setup(struct dsa_switch *ds)
+ 
+ 		/* Individual user ports get connected to CPU port only */
+ 		if (dsa_is_user_port(ds, i)) {
++			cpu_port = dp->cpu_dp->index;
++
+ 			ret = qca8k_rmw(priv, QCA8K_PORT_LOOKUP_CTRL(i),
```

Comment:
This looks wrong:
                if (dsa_is_user_port(ds, i)) {
                        int shift = 16 * (i % 2);
+                       cpu_port = dp->cpu_dp->index;

Reply:
why ?

Reply:
Patch doesn't apply

Reply:
I get an error an shall fix the patch. What is the sequnce to apply patches? I copy this patch to generic/hack-5.10 and give it number 996-34.

Reply:
The patch is not applying (hunk #3). If I understood @e9hack's comment correctly, the line with `int shift = 16 * (i % 2);` is required as the variable is used further down in the code.

However, if I correct the patch as per @e9hack's comment, the PR builds ok, the router works but I can't change ports with ip link set.